### ISSUE TITLE
[Feature request] Horizontal mouse scroll

### ISSUE THREAD

**Issue by migueltc13** [gnome-terminator/terminator#868]
**Problem:**
Terminator lacks support for horizontal mouse scrolling in TUI tools, impeding navigation within pagers (e.g., less) dealing with horizontally extensive content.

**Solution:**
Implement horizontal mouse scrolling in Terminator for seamless navigation in TUI applications with horizontal content and mouse support.

**Alternatives:**
Current alternatives involve reliance on keyboard shortcuts (akin to existing page_down/up).

**Additional Context:**
Evaluate scenarios within specific TUI tools (e.g., neovim, less) where horizontal scrolling would significantly improve usability.

**Comment by mattrose** [comment]
That needs to be implemented by vte, I think.  It seems like a fairly simple ask, but I don't see any work done on it.

Opened https://gitlab.gnome.org/GNOME/vte/-/issues/2765

**Comment by egmontkob** [comment]
Thanks @migueltc13 for this report and @mattrose for forwarding it!

It's a valid feature request, and unfortunately I missed it when occasionally checking Terminator's bugs.

**Comment by mattrose** [comment]
In record time, the above gitlab issue is fixed.  

Can you give me an exact example of how you would use this?  My thought was to use sudo journalctl -xe and try and horizontally scroll on the output of that, but maybe you have some other use cases?

**Comment by egmontkob** [comment]
`journalctl` calls `less` (or whatever your preferred viewer is), right?

I think it definitely deserves a feature request against `less`. Assuming it doesn't support it already. My version (590) doesn't seem to support it, and I couldn't find anything relevant in the changelogs since then, nor in the opened or closed bugs.

Similarly, any file viewer or editor could receive this feature.

I hope @migueltc13 has even more cool ideas. (I know you asked him, not me :))

**Comment by mattrose** [comment]
Yeah, I know journalctl calls less, but on my fedora install, it calls it in such a way that it doesn't wrap the lines, but lets them flow off the terminal to the left, and you can slide the window horizontally using arrow keys.

I was hoping that it would also interpret horizontal scrolling, but alas, no.  I might try neovim, but that means installing neovim and I'm worried it would interfere with my vim setup, as I use vim as my main editor.

**Comment by egmontkob** [comment]
`vim` supports horizontal scrolling, `neovim` doesn't (at least my versions). I used `vim` quite a lot for testing this feature.
